--- a/system/security/keystore/KeyStore.cpp
+++ b/system/security/keystore/KeyStore.cpp
@@ -27,6 +27,7 @@
 #include <utils/String8.h>
 
 #include <android-base/scopeguard.h>
+#include <android-base/properties.h>
 #include <android/hardware/keymaster/3.0/IKeymasterDevice.h>
 #include <android/security/IKeystoreService.h>
 #include <log/log_event_list.h>
@@ -601,6 +602,10 @@
     }
 
     auto version = getDevice(SecurityLevel::TRUSTED_ENVIRONMENT)->halVersion();
+    if (android::base::GetBoolProperty("ro.z1.stock_kernel", false) &&
+            version.securityLevel == SecurityLevel::SOFTWARE) {
+        return false;
+    }
     if (keyType == kRsaKeyType) return true;  // All versions support RSA
     return keyType == kEcKeyType && version.supportsEc;
 }
